File: UI8/JS/stack.js
function Stack() {
    let elements = [];
    let length = 0;

    this.push = function(element) {
        elements.push(element);
        length++;
    }

    this.pop = function() {
        if (elements.length !== 0) {
            length--;
            return elements.pop();
        }
        return null;
    }

    this.getStack = function() {
        return [...elements];
    }

    this.getLength = function() {
        return length;
    }
}

let stack = new Stack();
stack.pop();
stack.push(10);
stack.push(20);
console.log(stack.getStack());
stack.pop();
console.log(stack.getStack());

File: DSA_Assignment2/src/Stack/Stack.java
package Stack;

public class Stack {

	private static final int SIZE_OF_STACK = 100;
	private String[] stack = new String[SIZE_OF_STACK];

	//pointing to the latest added element of stack
	private int top = -1;


	public void push(String element) throws Exception {

		if (top==(SIZE_OF_STACK-1)){
			throw new Exception ("Stack is full");
		}
		top = top+1;
		stack[top] = element;
	}


	public String pop() throws Exception {

		if (top==-1){
			throw new Exception ("Stack is empty");
		}
		String poppedElement = stack[top];
		top = top-1;
		return poppedElement;
	}


	public Boolean isEmpty(){
		if (top==-1){
			return true;
		}
		return false;
	}


	public String peek() throws Exception{
		if (top==-1){
			throw new Exception ("Stack is empty");
		}
		return stack[top];
	}
}
